src/components/generic-link-share/component.tsx

```diff
@@ -9,14 +9,19 @@ import {
   PluginApi,
   LayoutPresentatioAreaUiDataNames,
   UiLayouts,
+  CurrentPresentation,
   RESET_DATA_CHANNEL,
+  pluginLogger,
+  PresentationToolbarButton,
 } from 'bigbluebutton-html-plugin-sdk';
 
 import GenericComponentLinkShare from '../generic-component/component';
 
 import { DataToGenericLink, DecreaseVolumeOnSpeakProps } from './types';
 import { ModalToShareLink } from '../modal-to-share-link/component';
 
+const SEARCH_PATTERN = /link(\s*\{(.|\n|\r\n)*\})/g;
```

Comment:
Please, put the regex within the body of the component. The problem here is that a regex variable stores last index from the search, and since this variable's scope is higher than the component's, it will store the last index and thus will return null in another iteration. See documentation here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#return_value